### PR TITLE
Add CI for notebooks

### PR DESCRIPTION
**Pull request by dominiquesydow**
## Description
Add CI to this repository that checks notebooks!

## Todos
  - [x] Rerun notebooks: Fix `klifs_utils` related errors (deprecated package) and replace with `opencadd.databases.klifs` instead
  - [x] Add `nbval` checks to notebooks
  - [x] Set up notebook CI (following @jaimergp's setup for [TeachOpenCADD](https://github.com/volkamerlab/teachopencadd/blob/master/.github/workflows/ci.yml))
    - [x] Check notebooks 1*, 2* and 3* first
    - [x] Check notebooks 4* separately (needs input that lives not in GH repository)
    - [x] Fix CI fail in notebook 4_3
    - [x] Split external data download and tests into two runs (suggested by @jaimergp)
    - [x] External data: Download and untar in one step: https://www.howtogeek.com/howto/uncategorized/linux-quicktip-downloading-and-un-tarring-in-one-step/ (suggested by @jaimergp)
    - [ ] ~External data: Add external data to GHA cache: https://github.com/actions/cache (suggested by @jaimergp)~ > _Not in this PR_
  - [x] Update GH TOC figure :)

## Questions
  - [x] How to test 4* notebooks? Input does not live in GH repository. > Usage of `wget` works as expected in `ci.yml` :)

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by dominiquesydow** [comment]
Celebrating first passing CI :partying_face: 

Now include talktorials 4* that need external data to be downloaded to the repo first.

**Comment by dominiquesydow** [comment]
Update: CI passes with Python 3.8, while fails with Python 3.9 in notebook 4_3 with an `StopIteration` error in context of `next` usage. Will now run _all_ notebooks under Python versions 3.6, 3.7, 3.8 and 3.9.

**Comment by dominiquesydow** [comment]
Changed my mind. Will run _all_ notebooks under Python 3.8 - if that works, pin Python version in environment to 3.8 (this repo does not contain a package but a collection of notebooks, so no need to check multiple Python versions for now).

**Comment by jaimergp** [comment]
Those `StopIteration` errors seem to come from a "no matches found" situation in your generator, I don't think it has to do with Python versions. You might have found a stochastic behaviour when creating molecules, though (a common cause for these "no matches found" errors is that the molecule object was empty to begin with).

**Comment by dominiquesydow** [comment]
Thanks, @jaimergp, for commenting on this!
After the Python 3.8 CI did fail this time, I also went back to thinking there is nothing to iterate over. I cannot reproduce this behavior locally.

Try out:
Test output of `exact_matches` variable (`nbval` check)
https://github.com/volkamerlab/KinFragLib/blob/master/notebooks/4_3_combinatorial_library_comparison_klifs.ipynb (cell 10)

If something is off with that one, constructing ligands might cause errors?
```python
Draw.MolsToGridImage(
    mols=[construct_ligand(row.fragment_ids, row.bond_ids, fragment_library) for index, row in exact_matches.iterrows()], 
    legends=[f'{row.group} | {row.kinase} | {row.complex_pdb} | {row.ligand_pdb}' for index, row in exact_matches.iterrows()],
    molsPerRow=6
)
```

**Comment by jaimergp** [comment]
Maybe print those comprehensions in a sep list?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+      - name: Run tests
+        shell: bash -l {0}
+        run: |
+          echo "Download combinatorial library from zenodo..."
+          wget -q -O data/combinatorial_library/combinatorial_library.tar.bz2 https://zenodo.org/record/3956580/files/combinatorial_library.tar.bz2?download=1
+          echo "Decompress combinatorial library..."
+          tar -xvf data/combinatorial_library/combinatorial_library.tar.bz2 -C data/combinatorial_library/
+          echo "Run tests..."
+          PYTEST_ARGS="--nbval-lax --current-env --dist loadscope --numprocesses 2"
```
Comment (jaimergp) [inline .github/workflows/ci.yml:67]:
```suggestion
          PYTEST_ARGS="--nbval-lax --current-env --nbval-cell-timeout=900"
```

Debug without paralellism for now (`--dist loadscope --numprocesses 2`), but put it back later if you can.

 --current-env         Force test execution to use a python kernel in the same
                        enviornment that py.test was launched from.

**Pull request merged**